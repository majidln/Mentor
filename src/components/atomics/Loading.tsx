import React, {FC, ReactElement} from 'react';
import {View, Text, StyleSheet} from 'react-native';

// TODO add spinkit loading
export const Loading: FC = (): ReactElement => {
  return (
    <View style={styles.wrapper}>
      <Text>Loading ....</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
  },
});
