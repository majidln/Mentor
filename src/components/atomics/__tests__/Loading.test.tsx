import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';

import {Loading} from './../Loading';

describe('Test for atomic Loading', () => {
  it('Loading render correctly two text', () => {
    const testRenderer = renderer.create(<Loading />);
    const testInstance = testRenderer.root;

    const textElements = testInstance.findAllByType(Text);

    expect(textElements.length).toBe(1);
  });

  it('Loading Snapshot renders correctly', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
