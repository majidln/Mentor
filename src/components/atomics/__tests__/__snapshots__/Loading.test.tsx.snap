// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for atomic Loading Loading Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text>
    Loading ....
  </Text>
</View>
`;
